taliforth.asm

```diff
@@ -543,6 +543,24 @@ underflow_4:
                 rts
 
 underflow_error:
+        ; """Display return stack to help find where underflow occurred"""
+                lda #<es_underflow2
+                sta tmp3
+                lda #>es_underflow2
+                sta tmp3+1
+                jsr print_common
```

Comment:
I'd prefer to have the Data Stack Underflow message printed first and then the return stack dump.  That will require restructuring things here a bit, as you won't be able to fall into `error` anymore, but it's probably worth it.  It might also make sense to pull the error string printing code out so that you can call it a couple of times from your routine and adjust `error` to use it as well.

I'll let the idea of a stack trace with name lookups rattle around in my head for a while more, but I think this is a good step in a direction that makes it easier for newcomers to use Tali.